platform/database/database.go

```diff
@@ -134,6 +135,15 @@ func IsForeignKeyConstraintError(err error, constraintName string) bool {
 	return e.Code == "23503" && e.Constraint == constraintName
 }
 
+// IsEnumInvalidValueError is a helper checking a database error and returns true if it's a invalid input value for an enum type
+func IsEnumInvalidValueError(err error) bool {
+	e, ok := err.(*pq.Error)
+	if !ok {
+		return false
+	}
+	return e.Code == "22P02" && strings.Contains(e.Message, "invalid input value for enum")
```

Comment:
I would really only check on the code and not care about the message here. I think in error messages a code is meant for consistency and a message is just for extra information so I think they'll be less likely to change the code than the message.

Reply:
not sure since the error with this code is only `invalid_text_representation`

Reply:
`22P02` is a generic error code for invalid input type errors, after digging through all the fields in the `*pq.error`, checking the message is the only way to know it's tied to an invalid enum value.